src/mds/mdstypes.cc

```diff
@@ -285,12 +286,15 @@ void inode_t::encode(bufferlist &bl, uint64_t features) const
   ::encode(last_scrub_version, bl);
   ::encode(last_scrub_stamp, bl);
 
+  if (features & CEPH_FEATURE_FS_BTIME)
+    ::encode(btime, bl);
```

Comment:
If we're doing conditional encode of the btime (based on features), we should set the encode version conditionally as well.
And actually, since the other end might be say an OSD which doesn't care, we should just encode unconditionally (especially since we're not forcing them to read the btime).